src/backend/utils/adt/network.c

```diff
@@ -389,6 +416,415 @@ network_cmp(PG_FUNCTION_ARGS)
+ * to save trips to the heap.
+ *
+ * +----------+-----------------------+--------------+--------------------+
+ * | 1 bit IP |    32 bits netmask    |    6 bits    |   25 bits subnet   |
+ * |  family  |        (full)         | netmask size |    (truncated)     |
+ * +----------+-----------------------+--------------+--------------------+
+ *
+ * IPv6
+ * ----
+ *
+ * 32-bit architecture:
+ *
+ * +----------+---------------------+
+ * | 1 bit IP |   31 bits netmask   |    (up to 97 bits
+ * |  family  |     (truncated)     |   netmask omitted)
+ * +----------+---------------------+
+ *
+ * 64-bit architecture:
+ *
+ * +----------+---------------------------------+
+ * | 1 bit IP |         63 bits netmask         |    (up to 65 bits
+ * |  family  |           (truncated)           |   netmask omitted)
+ * +----------+---------------------------------+
+ *
+ */
+static Datum
+network_abbrev_convert(Datum original, SortSupport ssup)
+{
+	network_sortsupport_state *uss = ssup->ssup_extra;
+	inet	   *authoritative = DatumGetInetPP(original);
+	Datum		res;
+	Datum		ipaddr_int,
+				netmask_int,
+				subnet_int;
+	char		datum_size_left,
+				datum_subnet_size;
+
+	/* TODO: Is this enough, or should the possibility of a third value be */
+	/* handled on branches below? */
+	Assert(ip_family(authoritative) == PGSQL_AF_INET ||
+		   ip_family(authoritative) == PGSQL_AF_INET6);
+
+	res = (Datum) 0;
+	if (ip_family(authoritative) == PGSQL_AF_INET6)
+	{
+		/* Shift a 1 over to the datum's most significant bit. */
+		res = ((Datum) 1) << SIZEOF_DATUM * BITS_PER_BYTE - ABBREV_BITS_INET_FAMILY;
+	}
+
+	/*
+	 * For IPv6, create an integer from either the first 4 or 8 bytes of the
+	 * IP (depending on whether we're on 32 or 64-bit architecture). For IPv4,
+	 * always use all 4 bytes.
+	 *
+	 * We're consuming an array of char, so make sure to byteswap on little
+	 * endian systems (an inet's IP array emulates big endian in that the
+	 * first byte is always the most significant).
+	 */
+	if (ip_family(authoritative) == PGSQL_AF_INET6)
+	{
+		ipaddr_int = *((Datum *) ip_addr(authoritative));
+		ipaddr_int = DatumBigEndianToNative(ipaddr_int);
+	}
+	else
+	{
+		uint32		ipaddr_int32 = *((uint32 *) ip_addr(authoritative));
+#ifndef WORDS_BIGENDIAN
+		ipaddr_int = pg_bswap32(ipaddr_int32);
+#endif
+	}
+
+	netmask_int = ipaddr_int;
+	subnet_int = (Datum) 0;
+	datum_subnet_size = 0;
+
+	/* Extract subnet bits from ipaddr_ints if there are some present. */
+	datum_size_left = SIZEOF_DATUM * BITS_PER_BYTE - ip_bits(authoritative);
+	if (datum_size_left > 0)
+	{
+		/*
+		 * Number of bits in subnet. e.g. An IPv4 that's /24 is 32 - 24 = 8.
+		 *
+		 * However, only some of the bits may have made it into the fixed
+		 * sized datum, so take the smallest number between bits in the subnet
+		 * and bits left in the datum.
+		 */
+		datum_subnet_size = Min(ip_maxbits(authoritative) - ip_bits(authoritative),
+								datum_size_left);
+
+		if (datum_subnet_size > 0)
```

Comment:
Need this branch? Can't you just shift all the bits away as a natural consequence of there being no room? Special cases should be avoided, as I mentioned.

Reply:
I ended up killing the branch.